### PR TITLE
Add support for MPPT metrics

### PR DESCRIPTION
**Pull request by sveba**
## Summary

* Adds MPPT (Maximum Power Point Tracking) metrics. MPPT metrics are archived by the fronius inverter every 5 mins.
* This results in an additional API request to the Fronius device with the same `--symo.timeout`
* Each API request is done sequentially. That means the total scrape time can be 2x `--symo.timeout`. Please take that into consideration when configuring Prometheus scrape interval
* Adds 2 new flags: `--symo.enable-power-flow`, `--symo.enable-archive` with default values being `true`. With these flags, you can also disable each API endpoint.

## Breaking Changes

* The `--symo.url` flag suggested to use the full URL path like `--symo.url http://symo.ip.or.hostname/solar_api/v1/GetPowerFlowRealtimeData.fcgi` previously. This has now changed to base host name only, e.g. `--symo.url http://symo.ip.or.hostname` (without trailing slash)

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `fix`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update documentation.
- [x] Update tests.

### CODE REVIEW

**Comment by ccremer** [comment]
Hi!
Thanks for the PR. I'll try to review it today or in the weekend.

**Review by ccremer — CHANGES_REQUESTED** [review]
Good work so far! 🎉 
I think it could be even more improved for both user and maintainers though.

File: metrics.go
Comment (sveba) [reply metrics.go:97]: Agree

File: metrics.go
Comment (sveba) [reply metrics.go:101]: Agree. will do

File: pkg/fronius/symo.go
Comment (sveba) [reply pkg/fronius/symo.go:114]: IMHO: I would prefer to not rely on array for mapping and referring the urls.  Thats why I created additional variable. Feel free to correct me, as I'm still very green in Go coming from JS and Java world :)

File: pkg/fronius/symo.go
Comment (sveba) [reply pkg/fronius/symo.go:123]: See previous comment

File: pkg/fronius/symo.go
Comment (sveba) [reply pkg/fronius/symo.go:154]: 1. Moving the logic out- OK
2. q.Set() was the first approach that I tried. The problem was the encoding after that. The fronius API didn't like the encoding of the date-params and delivered wrong timespan.  At least mine inverter. Maybe you ca try with yours.
3. The 5 mins timespan is hardcoded and the Fronius API delivers the values in 5 mins spans. There is the possibility to query for a bigger timespan and than the values map has multiple elements. Like this:
```
{
  "0": 200,
  "300": 212,
  "600": 140
  ...
}```
As this is prometheus exporter, we need only the value at the time of the scraping. Thats why I'm calculating the rounded timespan.

File: pkg/fronius/symo.go
Comment (sveba) [reply pkg/fronius/symo.go:114]: All parameters are necessary. About the values: I have only one inverter and one smart meter. My approach is always to cover the 95% of the use cases and if someone has additional different use case, than ask for test data (like json response) and work with it. 
But this is your baby, so you can decide the approach. I just don't feel comfortable to implement something based on assumptions :)

File: pkg/fronius/symo.go
Comment (ccremer) [reply pkg/fronius/symo.go:154]: 2) Ok. I would prefer it if you'd at least fmt.Sprintf() instead of string concatenation
3) Okay lets keep the 5 minutes.

File: pkg/fronius/symo.go
Comment (ccremer) [reply pkg/fronius/symo.go:114]: Hm, agreed arrays are kinda clumsy. I would be open for a map approach (prosa):
```go
var apiUrls = map[string]*url.URL{
  "powerflow": parseUrlOrDie("/solar_api/v1/GetPowerFlowRealtimeData.fcgi"),
  "archive": parseUrlOrDie("/solar_api/v1/GetArchi..."),
}

func parseUrlOrDie(url string) *url.URL {
u, err := url.Parse(options.URL + partial)
	    if err != nil {
		    panic(err)
	    }
	    return u
}

func ... {
  ...

		requestArchiveData: &http.Request{
			URL:    apiUrls["archive"],
}
```

Comment (sveba) [reply pkg/fronius/symo.go]: Great, had the same idea with the const. Otherwise a little bit different solution. I hope it fits your QA :)

**Review by ccremer — COMMENTED** [review]
2 things that came to mind: Right now the `timeout` option is applied to both requests. Meaning a timeout of 5 seconds could potentially end up in a scrape of 10 seconds. If we have further Data to scrape, 15 seconds etc. It should be at least documented what this means.

However, thinking even further, it's probably a good idea to make the requests in parallel. That should end up in faster scraping. A little more care needs to be given regarding error handling, in order to present only 1 error to the user, I'd propose to add https://github.com/hashicorp/go-multierror into the mix: Each fatal scrape error can be combined into one and it nicely displays multiple occurred errors. 

I would be fine with doing this in a second step in a separate PR. If you would be interested in doing this, I would also delay a release until we have parallelization.

File: pkg/fronius/symo.go
Comment (ccremer) [reply pkg/fronius/symo.go]: Yep, your solution is better than my braindump. I'm happy with that

File: pkg/fronius/symo.go
Comment (sveba) [reply pkg/fronius/symo.go:108]: Done

File: metrics.go
Comment (sveba) [reply metrics.go:129]: Done

File: cfg/types.go
Comment (sveba) [reply cfg/types.go:37]: done

File: Makefile
Comment (sveba) [reply Makefile:17]: accident, fixed

**Comment by sveba** [comment]
> I would be fine with doing this in a second step in a separate PR. If you would be interested in doing this, I would also delay a release until we have parallelization.

Agree, but additional PR would be my choice

File: Makefile
Comment (ccremer) [reply Makefile:17]: I'm not sure about that :see_no_evil: The latest diff on GH hasn't changed

Comment (sveba) [reply Makefile]: sorry about that. Now it should be fixed

**Review by ccremer — APPROVED** [review]
Thanks for the work!
Do you know if you are going to work on the parallelization right away or would you like a prerelease first?

**Comment by sveba** [comment]
> Thanks for the work!
> Do you know if you are going to work on the parallelization right away or would you like a prerelease first?

I'm leaving in vacation for 1,5 months so I would not be able to work on this for that time. After that no problem. 
About release: like I said, this is your baby, you can decide. I have the exporter running on my server so I'm not in a hurry. :)

**Comment by ccremer** [comment]
Ah ok. I'll wait for a release and try to come up with something, shouldn't be too hard.
Enjoy your vacation and thanks again for your contribution!

**Comment by sveba** [comment]
Na klar. Habe ich gerne gemacht. Wie gesagt, bin absoluter Go-Anfänger ;)

**Pull request merged**
